### PR TITLE
chore(release): Merge hotfix_6.2.1@c69fe0069 into edge (merge 1)

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

This PR merges the first handful of commits from the `hotfix_6.2.1` branch into `edge`. We're going incrementally; the remaining commits will be merged with separate follow-up PRs.

# Changelog

This first merge involves two manual resolutions:

## robot-server/robot_server/persistence/legacy_pickle.py

On `edge`, PR #11606 added `LabwareMovementStrategy` to the `_legacy_ot_types` list.

On `hotfix_6.2.1`, PR #11615 moved the `_legacy_ot_types` list from a global variable to a function.

My resolution is to keep the function and make sure it contains the `LabwareMovementStrategy` addition.

## app/src/organisms/Devices/__tests__/RobotOverviewOverflowMenu.test.tsx

On `edge`, these PRs touched this test file:

  * #11577
  * #11656
  * #11909

On `hotfix_6.2.1`, PR #11892 touched this same test file to support a change in production behavior.

I resolved this with @brenthagen.

# Test plan:

* [ ] Smoke-test this by running a dev server and a dev app.
* [x] Look closely to make sure all CI jobs are passing.

# Code review requests

* JS people: does the new `RobotOverviewOverflowMenu.test.tsx` look correct?
* Python people: does the new `legacy_pickle.py` look correct?

# Risk assessment

Low. This is a big diff, but nearly all of it has already been reviewed and tested. We just need to make sure the manual merge resolutions are correct, and smoke-test to make sure there were no botched auto-merges elsewhere.

### CODE REVIEW

**Comment by SyntaxColoring** [comment]
Reminder to self to *not* click the GitHub "Squash and merge" button. This needs to be a normal merge into `edge` to properly link up the commit DAG.

File: app/src/organisms/Devices/__tests__/RobotOverviewOverflowMenu.test.tsx [hunk 1]
```diff
@@ -184,17 +184,23 @@ describe('RobotOverviewOverflowMenu', () => {
...
+    getByRole('button', { name: 'Restart robot' })
+    getByRole('button', { name: 'Home gantry' })
+    getByRole('button', { name: 'Robot settings' })
+  })
+  it('should disable settings link when the robot is unreachable', () => {
+    const { getByRole } = render({ robot: mockUnreachableRobot })
+    const btn = getByRole('button')
+    fireEvent.click(btn)
     expect(getByRole('button', { name: 'Robot settings' })).toBeDisabled()
   })
```
Comment (b-cooper) [inline app/src/organisms/Devices/__tests__/RobotOverviewOverflowMenu.test.tsx:205]:
```suggestion
  it('should disable settings link when the robot is unreachable', () => {
    const { getByRole } = render({ robot: mockUnreachableRobot })
    const btn = getByRole('button')
    fireEvent.click(btn)
    expect(getByRole('button', { name: 'Robot settings' })).toBeDisabled()
  })
   it('should enable settings link when the robot is reachable', () => {
    const { getByRole } = render({ robot: mockReachableRobot })
    const btn = getByRole('button')
    fireEvent.click(btn)
    expect(getByRole('button', { name: 'Robot settings' })).not.toBeDisabled()
  })
```

Comment (SyntaxColoring) [reply app/src/organisms/Devices/__tests__/RobotOverviewOverflowMenu.test.tsx:205]: Ah—does this test already exist above, starting on line 155?

https://github.com/Opentrons/opentrons/blob/7558d3d37737d6577b5e7e507c211c2d6b93728a/app/src/organisms/Devices/__tests__/RobotOverviewOverflowMenu.test.tsx#L155-L162

**Review by b-cooper — APPROVED** [review]
👔

**Review by mcous — APPROVED** [review]
Python changes LGTM

**Pull request merged**